plugins/scaffolder-backend/src/scaffolder/templater/helpers.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Docker from 'dockerode';
+import { Writable, PassThrough } from 'stream';
+import fs from 'fs';
+
+export type RunDockerContainerOptions = {
+  imageName: string;
+  args: string[];
+  logStream?: Writable;
+  resultDir: string;
+  templateDir: string;
+};
+
+const dockerClient = new Docker();
```

Comment:
Maybe best to start passing this around straight away instead of a singleton?

Reply:
Yep, good call